ni_measurementlink_service/session_management/_types.py

```diff
@@ -320,3 +320,21 @@ class SessionInitializationBehavior(IntEnum):
     Note: When using the Session as a context manager and the context exits, it
     will detach from the server session and leave it open.
     """
+
+    INITIALIZE_SERVER_SESSION_THEN_DETACH = 3
+    """
+    Require the NI gRPC Device Server to initialize a new session with the
```

Comment:
If this is meant to work with third party drivers, we might want to remove references to "gRPC Device Server" from the help. Similarly, we should probably update the documentation to explain the "close behavior" only applies when used as a context manager. Otherwise, the user still has to call the driver close API directly with whatever behavior is desired. I don't recall if we were going to offer a generic `close_sessions` API directly from the reservation class like there is in LV. If there is going to be one, then that API won't have an input for the close behavior. Instead, the behavior specified when creating the session will be applied.

Reply:
I found the asymmetry between the enum value names a bit weird, but I see that you are just following what VISA has planned here. We'll have to decide how much we value consistency vs. symmetry here (or get VISA to change their plans to use symmetrical names.

Reply:
> I don't recall if we were going to offer a generic close_sessions API directly from the reservation class like there is in LV.

In Python, the `create_foo_session(s)` API will return the session as a context manager whose exit block will take care of the Session's close behavior internally. Hence, we won't be exposing any generic or driver-specific `close_sessions` API.

Reply:
This enum started out as a copy of the nimi-python one that we can share between drivers. That's why the docs refer to NI gRPC Device Server.

I don't mind changing the docs and enum values to be more general, if it makes sense.

```
ATTACH_TO_SERVER_SESSION = 3, # (This will attach and close)
``` 
I don't like this because it makes the atypical behavior (mixing attach and close) implicit.

The typical behavior is init/close or attach/detach.

The asymmetry calls out the fact that these two cases are atypical: init/detach or attach/close.

```
INITIALIZE_SESSION_AUTO_CLOSE
```
These make it harder to distinguish between the typical and atypical cases.

Reply:
For now, I've updated the atypical enum values alone with Joel's suggestions. If everyone agrees with the naming, I'll update the existing enum values with the approved suggestions.

Reply:
> If this is meant to work with third party drivers, we might want to remove references to "gRPC Device Server" from the help. Similarly, we should probably update the documentation to explain the "close behavior" only applies when used as a context manager.

@jasonmreding I think this is resolved.